File: TaskBuilder/Models/Function/IFunctionModel.cs
using System;
using System.Collections.Generic;

namespace TaskBuilder.Models.Function
{
    public interface IFunctionModel
    {
        Guid TypeGuid { get; }

        string DisplayName { get; }

        string DisplayColor { get; }

        IInvokeModel Invoke { get; }

        ICollection<IDispatchModel> Dispatchs { get; }

        ICollection<IInputModel> Inputs { get; }

        ICollection<IOutputModel> Outputs { get; }
    }
}

File: TaskBuilder.React/models/function/IFunctionModel.ts
import { IInvokeModel } from "./IInvokeModel";
import { IDispatchModel } from "./IDispatchModel";
import { IInputModel } from "./IInputModel";
import { IOutputModel } from "./IOutputModel";

export interface IFunctionModel {
    typeGuid: string;
    displayName: string;
    displayColor: string;
    invoke: IInvokeModel;
    dispatchs: IDispatchModel[];
    inputs: IInputModel[];
    outputs: IOutputModel[];
}